libnetwork/networkdb/cluster_test.go

```diff
@@ -0,0 +1,167 @@
+package networkdb
+
+import (
+	"maps"
+	"math"
+	"math/bits"
+	"slices"
+	"strings"
+	"testing"
+
+	"github.com/montanaflynn/stats"
+	"pgregory.net/rapid"
+)
+
+func TestMRandomNodes(t *testing.T) {
+	cfg := DefaultConfig()
+	// The easiest way to ensure that we don't accidentally generate node
+	// IDs that match the local one is to include runes that the generator
+	// will never emit.
+	cfg.NodeID = "_thisnode"
+	uut := newNetworkDB(cfg)
+
+	t.Run("EmptySlice", func(t *testing.T) {
+		sample := uut.mRandomNodes(3, nil)
+		if len(sample) != 0 {
+			t.Errorf("got sample size %d, want 0", len(sample))
+		}
```

Comment:
any reason for not using the `assert` package?


```suggestion
        assert.Check(t, is.Len(sample, 0))
```

(here and in other places)

Reply:
Personally I wouldn't mind such inconsistency. The `gotest.tools`'s assert are easier to read and falling back to imperative checks for more complex checks isn't bad IMO.